File: crsort/crsort.js
function f(L) {
    let n;
    let len = L.length;
    let m = 1;
    let sl = 1;
    let tmp;
    while (m < len) {
        n = 0;
        while (n < sl) {
            if (L[m] < L[n]){
                tmp = L[m];
                g = m;
                while (g > n) {
                    L[g] = L[g-1];
                    g --;
                }
                L[n] = tmp;
                sl ++;
                break;
            }
            n ++;
        }
        m ++;
    }
}

function main() {
    let L = [];
    let i = 1;
    let size = 1000;
    while (i <= size) {
        L.push(i);
        i ++;
    }
    L.reverse();
    f(L);
    console.log(L);
}

main();

// time 0.086

File: crsort/crsort.py
def f(l):
    n = 0
    m = 1
    sl = 1

    while m < len(l):
        n = 0
        while n < sl:
            if l[m] < l[n]:
                g = m
                tmp = l[m] 
                while g > n:
                    l[g] = l[g-1]
                    g -= 1
                l[n] = tmp
                sl += 1
                break
            n += 1        
        m += 1
    return l

L = [x for x in range(1000)]
L.reverse()
print(f(L))

#time 0.124
